addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -135,18 +135,16 @@ def get_vat_country(vat):
                 return 'IT'
             return vat[:2].upper()
 
-        def in_eu(partner):
-            europe = self.env.ref('base.europe', raise_if_not_found=False)
-            country = partner.country_id
-            if not europe or not country or country in europe.country_ids:
-                return True
-            return False
-
         def format_alphanumeric(text_to_convert):
             return text_to_convert.encode('latin-1', 'replace').decode('latin-1') if text_to_convert else False
 
+        def get_partner(x):
+            return x if (x is None or x._name != 'res.company') else x.partner_id
```

Comment:
The thing is, it can be that
`buyer == commercial_partner` and `seller == company`
or `buyer == company` and `seller == commercial_partner` 
The second is in case of import, we have to send through EDI a simulation of the original invoice that we were sent by the EU partner, as a "self-invoice" in which we are the buyer, and the partner is the seller.

So if I use `buyer.<field_that_partners_have_but_companies_dont>` it tracebacks for self-invoices.

That's why I'm differentiating between the `buyer` and the `buyer_partner`.

But I agree that I can do it in the code instead of passing the function